### PR TITLE
[SPARK-16154] [MLLIB] Update spark.ml and spark.mllib package docs

### PR DESCRIPTION
**Pull request by mengxr**
## What changes were proposed in this pull request?

Since we decided to switch spark.mllib package into maintenance mode in 2.0, it would be nice to update the package docs to reflect this change.
## How was this patch tested?

Manually checked generated APIs.

### CODE REVIEW

**Comment by mengxr** [comment]
cc: @jkbradley @MLnick @srowen

**Comment by jkbradley** [comment]
LGTM pending tests

File: mllib/src/main/scala/org/apache/spark/mllib/package-info.java [hunk 1]
```diff
@@ -16,6 +16,26 @@
  */
 
 /**
- * Spark's machine learning library.
+ * RDD-based machine learning APIs (in maintenance mode).
+ *
+ * The <code>spark.mllib</code> package is in maintenance mode as of the Spark 2.0.0 release to
+ * encourage migration to the DataFrame-based APIs under the <code>spark.mllib</code> package.
```
Comment (hhbyyh) [inline mllib/src/main/scala/org/apache/spark/mllib/package-info.java]: is it <code>spark.ml</code> ?

**Comment by MLnick** [comment]
Looks good subject to @hhbyyh comment above

File: mllib/src/main/scala/org/apache/spark/mllib/package-info.java (hunk 1 above)
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/package-info.java]: Ah, I made this typo while playing with JavaDoc tags ... I tried to link to the `spark.ml` Java package doc but no success and ended up with `<code>`.

**Comment by mengxr** [comment]
Merged into master and branch-2.0.

**Pull request closed without merging**